File: persistenece/src/main/java/com/anokmik/persistence/model/TripEvent.java
package com.anokmik.persistence.model;

import com.anokmik.persistence.TripAssistantDatabase;
import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.ForeignKey;
import com.raizlabs.android.dbflow.annotation.ForeignKeyReference;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.structure.BaseModel;

@Table(database = TripAssistantDatabase.class)
public final class TripEvent extends BaseModel {

    @PrimaryKey(autoincrement = true)
    public long id;

    @ForeignKey(references = {
            @ForeignKeyReference(
                    columnType = long.class,
                    columnName = "trip",
                    foreignKeyColumnName = "id"
            )
    })
    @Column
    public Trip trip;

    @ForeignKey(references = {
            @ForeignKeyReference(
                    columnType = long.class,
                    columnName = "user",
                    foreignKeyColumnName = "id"
            )
    })
    @Column
    public User user;

    @Column
    public String name;

    @Column
    public String comment;

    @Column
    public long startDate;

    @Column
    public long finishDate;

}

File: persistenece/src/main/kotlin/com/anokmik/persistence/model/TripEvent.kt
package com.anokmik.persistence.model

import android.support.annotation.IntDef
import com.anokmik.persistence.TripAssistantDatabase
import com.raizlabs.android.dbflow.annotation.*
import com.raizlabs.android.dbflow.structure.BaseModel

@Table(database = TripAssistantDatabase::class)
class TripEvent : BaseModel() {

    @JvmField
    @PrimaryKey(autoincrement = true)
    var id: Long = 0

    @JvmField
    @ForeignKey(references = arrayOf(
            ForeignKeyReference(
                    columnType = Long::class,
                    columnName = "trip",
                    foreignKeyColumnName = "id"
            )
    ))
    @Column
    var trip: Trip? = null;

    @ForeignKey(references = arrayOf(
            ForeignKeyReference(
                    columnType = Long::class,
                    columnName = "user",
                    foreignKeyColumnName = "id"
            )
    ))
    @Column
    var user: User? = null;

    @JvmField
    @Column
    var name: String? = null

    @JvmField
    @Column
    var comment: String? = null

    @JvmField
    @Column
    var startDate: Long? = 0

    @JvmField
    @Column
    var finishDate: Long? = 0


}